[IMP] Inventory: Bringing reassign lot/serial numbers current (#15916)
odoo/documentation: content/applications/inventory_and_mrp/inventory/product_management/product_tracking/reassign.rst

Changing a product's tracking settings to use lots or serial numbers, *after* storing products in
Odoo without them, can lead to inconsistent records. Follow this documentation to learn how to use

> an inventory adjustment to assign lot or serial numbers to products that were not originally

Reviewer: ```suggestion
an inventory adjustment to assign lot or serial numbers to products that were **not** originally
```

Reviewer: I definitely missed the entire concept of this happening ONLY to items WITHOUT any lots/numbers. I can't believe I missed that! So this is just to help the other "me's" out there =D

assigned any.

To enable lots and serial numbers, go to :menuselection:`Inventory app --> Configuration -->
Settings`, then select :guilabel:`Lots & Serial Numbers`.

.. note::
   This document outlines the process of using two inventory adjustments: one to remove incorrect
   records *without* lot numbers, and another to save the quantities *with* the lot numbers.

.. important::

>    :guilabel:`Track Inventory` must be enabled for the product to follow the process outlined in

Reviewer 2: minor style thing (: Since the "Track Inventory" field isn't on the page until *after* this section, let's leave this at italics for now.

```suggestion
   *Track Inventory* must be enabled for the product to follow the process outlined in
```

   this document. It is not possible to add inventory tracking by lot or serial number if the

>    product is untracked and then purchased or sold. Only unused products can be tracked.

Reviewer: ```suggestion
   Once an item is purchased from inventory, that product **cannot** be tracked. Only *unsold* inventoried items can have trakcing enabled, and have lots or serial numbers added. 
   
   Additionally, the :guilabel:`Track Inventory` option **must** be enabled for the product to follow the process outlined in
   this document. 
```

I just wanted to start with the ONCE X HAPPENS you can't do Y - since I missed that at first.  This is just a suggestion.

And I hate to be a super nit-picke,r BUT if you did NOT have the 'track Inventory' option initially selected/enabled for a product you have stoed (but NOT sold yet) - I assume you can start to track it, and then reassign the lots/SN's? I don't think it's a common case, but if that IS possible I mightmention it. You can even start with the below first paragraph, to 'activate the option X if not already activated'. 

Again, probably not common, but I always think aotu every scenario =). 

And since it's sucn an outlier you can ignore this!

an inventory adjustment to assign lot or serial numbers to products that were not originally

> assigned lots.

Reviewer: ```suggestion
assigned any.
```
or "them" or "lots or serial numbers" Just because you say 'lots or serial numbers' right befor,e then only mention 'lots'.

.. note::
   This document outlines the process of using two inventory adjustments: one to remove incorrect

>    records *without* lot numbers, and another to save the quantities *with* the lot numbers.

Reviewer: I don't know if this is possible, but this could be pulled out of the note and just added to the main paragraph above:

"The method of assigning lots and serial numbers invovles two steps: A and B."

Then, if ther eis a way to link the step (A and B) to their respective steps below, may be helpful. I only started reading this doc so that may not be possible!

   - :doc:`Set up and use lot numbers <lots>`
   - :doc:`Use serial numbers <serial_numbers>`

.. important::
   :guilabel:`Track Inventory` must be enabled for the product to follow the process outlined in
   this document. It is not possible to add inventory tracking by lot or serial number if the
   product is untracked and then purchased or sold. Only unused products can be tracked.

Change product tracking and perform inventory adjustments
=========================================================

To change the product's settings to track by lots or serial numbers, begin by navigating to
:menuselection:`Inventory app --> Products --> Products`, and select the intended product.

Change the :guilabel:`Track Inventory` value to :guilabel:`By Lots` or :guilabel:`By Unique Serial

> Number`. A message appears that explains that an inventory adjustment is required to add lot or

Author: This setting is actually mentioned above this section in an important block. These instructions are for products that do not have lots or serial numbers applied, or if the lots and serial numbers are incorrect.

serial numbers to existing inventory.

> .. image:: reassign/warning.png

Reviewer: I dont know if this image is necessary - and it's not as clear/is more blurry than the one below it, so I would sugest removing it.

>    :alt: Warning message: products in stock have no lot/serial number.

Author: This is expected. Until 19, you couldn't add lots/serial numbers to product that was previously purchased/sold in inventory.

I did mention in an important block above this entire section (up at the top) that tracking needs to be enabled before a product is "used" (purchased/sold) in order for this process to work. It seems like the only way to work around this is to create a duplicate product (manually) in the database and start tracking it that way, then add inventory through this process here and remove it from the other listing.

> Hover over the :guilabel:`Quantity On Hand` field. An :guilabel:`Update` link appears.

Reviewer: ```suggestion
Hover over the :guilabel:`Quantity On Hand` field and an :guilabel:`Update` link appears.
```
The two sentences just looked so short =D

> .. image:: reassign/update-link.png

Reviewer: !!!! Again, I feel like I must have not had my head screwed on right, because I swore I looked and didn't see it. I am so sorry.

   :alt: The Update link appears to the right of the Quantity On Hand field.

Click the :guilabel:`Update` link. The :guilabel:`Update Quantity` page opens.

Update the :guilabel:`On Hand Quantity` field to `0`. Click the :guilabel:`Save` button.

.. note::
   If the product is stored in multiple locations, make sure the **total** on hand quantity at
   **all** locations is zero.

.. image:: reassign/remove-quant.png
   :alt: The Update Quantity page, highlighting the On Hand Quantity field.

Then, click the :guilabel:`New` button. Add a lot or serial number in the :guilabel:`Lot/Serial
Number` field, and specify a quantity. Click the :guilabel:`Save` button. Repeat this process for
all lots or serial numbers for the product in the inventory.

.. image:: reassign/update-quantity.png
   :alt: Fill in the Lot/Serial Number and On Hand Quantity fields.

.. seealso::
   :doc:`../../warehouses_storage/inventory_management/count_products`

.. tip::
   To find the original quantity, and adjust the :guilabel:`On Hand Quantity` accordingly, after
   assigning a new lot or serial number, click the :icon:`fa-history` :guilabel:`History` button on

>    the far-right.

Author: So it turns out you want the first History link for the line you edited the first time, so I attempted to clarify that in my update that I'm about to add here. Let me know if it could be clearer.

Even if the user adds a line with the serial number, then clicks the History link, then returns to the Update Quantity page, they can still edit the quantity on the line they added with a serial number.

Let me know if that's confusing. (I had a hard time writing that sentence, so it wouldn't surprise me!)
